### DISCUSSION TITLE
Your Search Console account hasn't been connected successfully.

### DISCUSSION THREAD

**Discussion by jesperordrup** [plausible/analytics#3008] [Self-Hosted Support]
Any time I try to connect to search console I get the above error.

The account I connect to have 5 verified websites (properties). All working fine. But when I try to connect in plausible, i get this:

![image](https://github.com/plausible/analytics/assets/3120151/306f755f-6f4f-4c4f-8af0-3bc374794c59)

**Comment by ruslandoga** [comment]
👋 @jesperordrup 

I'm not familiar with that part of the app, but maybe checking the logs would show some insight. Or, if not, I remember there are some [places](https://github.com/search?q=repo%3Aplausible%2Fanalytics%20capture_&type=code) where logs are suppressed and the error is sent to Sentry instead. So it might make sense to add `SENTRY_DSN` env var to catch all errors. That would most likely show what the problem is.

**Reply by ruslandoga** [reply]
[Tests](https://github.com/plausible/analytics/blob/995556e86f7faf420d4c18db6721d532eb1f45f2/test/plausible_web/controllers/site_controller_test.exs#L527-L545) and [implementation](https://github.com/plausible/analytics/blob/995556e86f7faf420d4c18db6721d532eb1f45f2/lib/plausible/google/http.ex#L96-L114) of the relevant function might also provide some insights.

**Reply by ruslandoga** [reply]
In short, that error message seems to be shown when `https://www.googleapis.com/webmasters/v3/sites` returns 401 or 403 status codes.

<details><summary>why that might happen, according to ChatGPT...</summary><br>

The 401 and 403 status codes from the HTTP protocol stand for "Unauthorized" and "Forbidden", respectively. If you're getting these status codes from the Google Search Console API endpoint, it typically suggests a problem with your authentication or authorization.

1. HTTP 401 Unauthorized: This status code is returned when the request lacks valid authentication credentials for the target resource. If you're receiving this, it means that you did not provide the correct API key or OAuth 2.0 token for the request. Make sure you've properly authenticated with OAuth 2.0 and obtained the correct access token. 

2. HTTP 403 Forbidden: This status code is returned when the client does not have permission to access the requested resource, even with a valid set of credentials. This usually means the authenticated user doesn't have the necessary permissions for a resource or for the specific actions you are trying to execute. If you are receiving this error, it may mean that the authenticated user doesn't have permission to view the requested site's data in Google Search Console.

In both cases, you'll need to check and ensure that you've properly set up your API credentials, authenticated your application, and ensured the authenticated user has the necessary permissions in Google Search Console to perform the desired action. In case of using OAuth 2.0, also make sure that your token has not expired.

</details

**Reply by jesperordrup** [reply]
> [Tests](https://github.com/plausible/analytics/blob/995556e86f7faf420d4c18db6721d532eb1f45f2/test/plausible_web/controllers/site_controller_test.exs#L527-L545) and [implementation](https://github.com/plausible/analytics/blob/995556e86f7faf420d4c18db6721d532eb1f45f2/lib/plausible/google/http.ex#L96-L114) of the relevant function might also provide some insights.

Cant - we have installed a release - I have no real dev setup for this.

**Reply by jesperordrup** [reply]
> 👋 @jesperordrup
> 
> I'm not familiar with that part of the app, but maybe checking the logs would show some insight. Or, if not, I remember there are some [places](https://github.com/search?q=repo%3Aplausible%2Fanalytics%20capture_&type=code) where logs are suppressed and the error is sent to Sentry instead. So it might make sense to add `SENTRY_DSN` env var to catch all errors. That would most likely show what the problem is.

Ill try this

**Reply by jesperordrup** [reply]
Ive added a SENTRY_DNS and rebooted and retried. There's nothing sent to sentry. Do I need to do anything else?

**Reply by ruslandoga** [reply]
Sorry, according to https://github.com/plausible/analytics/discussions/3008#discussioncomment-6100394 it doesn't seem that anything is being sent to Sentry in case of an auth error. But the error is kind of known, it's https://github.com/plausible/analytics/discussions/3008#discussioncomment-6100399

> Cant - we have installed a release - I have no real dev setup for this.

I meant that you can read the code I linked to better understand the auth flow and what might cause the problem you are seeing.

**Reply by ruslandoga** [reply]
> There's nothing sent to sentry. Do I need to do anything else?

You can try running something like `Sentry.capture_message("test")` from the console to check the integration.

```console
$ cd hosting
$ docker compose exec plausible bin/plausible remote
iex> Sentry.capture_message("test")
```

**Reply by jesperordrup** [reply]
Test works fine:

![image](https://github.com/plausible/analytics/assets/3120151/b9fcd8b1-386c-48e6-9bb5-6d1210b36248)

**Comment by BlaineMoore** [comment]
Have you managed to determine what the problem is? I'm getting the same message, I've unlinked and relinked a few times, and the permissions etc all seem to be setup, everything looks like it is working until I return to plausible and it tells me something went wrong without any further details.

**Reply by jesperordrup** [reply]
Sorry no. I've tried the suggestions but no luck. There's nothing in logs so I cant proceed. I've tried to create an issue #3031 but it was closed.

**Reply by adanselm** [reply]
Have you enabled the Google Search Console API from the console here? https://console.cloud.google.com
"Enabled APIs and Services" > Click the "+ ENABLE APIS AND SERVICES" at the top and search for "search console".
